agent/api/container/container.go

```diff
@@ -1360,3 +1367,39 @@ func (c *Container) IsContainerTornDown() bool {
 	defer c.lock.RUnlock()
 	return c.ContainerTornDownUnsafe
 }
+
+func (c *Container) SetContainerHasPortRange(containerHasPortRange bool) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.ContainerHasPortRange = containerHasPortRange
+}
+
+func (c *Container) GetContainerHasPortRange() bool {
```

Comment:
Question; I'm wondering if these reads need locks as well. Depending on how often this data changes, we might be able to get away without the reads being locked. Non blocking though.

Reply:
the get methods take a read lock, which multiple goroutines can acquire at the same time. Also, keeping it consistent with other getters in this file